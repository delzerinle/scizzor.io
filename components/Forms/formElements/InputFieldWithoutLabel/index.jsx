import PropTypes from 'prop-types';
import { Field, ErrorMessage } from 'formik';

const InputFieldWithoutLabel = ({
  name,
  label,
  icon,
  type = 'text',
  errors,
  touched,
  fieldClassName = 'border-alt-2',
}) => {
  const [parentKey, childKey] = name.split('.');

  const wrapperClass = icon
    ? `flex border w-full ${fieldClassName} field-input ${
        errors?.[parentKey]?.[childKey] && touched?.[parentKey]?.[childKey]
          ? 'mb-10 field-error'
          : 'mb-4'
      }`
    : `border w-full ${fieldClassName} field-input ${
        errors?.[parentKey]?.[childKey] && touched?.[parentKey]?.[childKey]
          ? 'mb-10 field-error'
          : 'mb-4'
      }`;

  const inputWrapperClass = icon
    ? 'flex-1 px-4 py-2 relative -ml-4'
    : 'px-4 py-2 relative';

  return (
    <div className={wrapperClass}>
      {icon && <div className="px-4 flex items-center">{icon}</div>}
      <div className={inputWrapperClass}>
        <Field
          id={name}
          type={type}
          name={name}
          placeholder={label}
          className="w-full focus:outline-none font-light text-sm placeholder-secondary bg-transparent"
        />
        <ErrorMessage
          name={name}
          component="p"
          className="p-sm absolute text-secondary pt-2"
        />
      </div>
    </div>
  );
};

InputFieldWithoutLabel.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.node,
  errors: PropTypes.object,
  touched: PropTypes.object,
  fieldClassName: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default InputFieldWithoutLabel;
