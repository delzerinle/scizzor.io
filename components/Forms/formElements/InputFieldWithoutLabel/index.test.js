import React from 'react';
import { shallow } from 'enzyme';

import InputFieldWithoutLabel from './index';

describe('InputFieldWithoutLabel tests', () => {
  it('renders without crashing', () => {
    shallow(
      <InputFieldWithoutLabel
        name="test"
        label="test"
        errors={{ test: 'test' }}
        touched={{ test: 'test' }}
      />
    );
  });

  it('input field snapshot', () => {
    const wrapper = shallow(
      <InputFieldWithoutLabel
        name="test"
        label="test"
        errors={{ test: 'test' }}
        touched={{ test: 'test' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('input field with icon snapshot', () => {
    const wrapper = shallow(
      <InputFieldWithoutLabel
        name="test"
        label="test"
        icon="test"
        errors={{ test: 'test' }}
        touched={{ test: 'test' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
