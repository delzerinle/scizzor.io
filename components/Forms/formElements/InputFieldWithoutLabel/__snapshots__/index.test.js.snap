// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldWithoutLabel tests input field snapshot 1`] = `
<div
  className="border w-full border-alt-2 field-input mb-4"
>
  <div
    className="px-4 py-2 relative"
  >
    <Field
      className="w-full focus:outline-none bg-transparent font-light text-sm placeholder-secondary"
      id="test"
      name="test"
      placeholder="test"
      type="text"
    />
    <FormikConnect(ErrorMessageImpl)
      className="p-sm absolute text-secondary pt-2"
      component="p"
      name="test"
    />
  </div>
</div>
`;

exports[`InputFieldWithoutLabel tests input field with icon snapshot 1`] = `
<div
  className="flex border w-full border-alt-2 field-input mb-4"
>
  <div
    className="px-4 flex items-center"
  >
    test
  </div>
  <div
    className="flex-1 px-4 py-2 relative -ml-4"
  >
    <Field
      className="w-full focus:outline-none bg-transparent font-light text-sm placeholder-secondary"
      id="test"
      name="test"
      placeholder="test"
      type="text"
    />
    <FormikConnect(ErrorMessageImpl)
      className="p-sm absolute text-secondary pt-2"
      component="p"
      name="test"
    />
  </div>
</div>
`;
